google-http-client-apache-v5/src/main/java/com/google/api/client/http/apache/v5/Apache5ResponseContent.java

```diff
@@ -0,0 +1,41 @@
+package com.google.api.client.http.apache.v5;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+
+/**
+ * Class that wraps an {@link org.apache.hc.core5.http.HttpEntity}'s content {@link InputStream}
+ * along with the {@link ClassicHttpResponse} that contains this entity. The main purpose is to be
+ * able to close the response as well as the content input stream when {@link #close()} is called.
+ */
+public class Apache5ResponseContent extends InputStream {
+  private final ClassicHttpResponse response;
+  private final InputStream wrappedStream;
+
+  public Apache5ResponseContent(InputStream wrappedStream, ClassicHttpResponse response) {
+    this.response = response;
+    this.wrappedStream = wrappedStream;
+  }
+
+  @Override
+  public int read() throws IOException {
+    return wrappedStream.read();
+  }
+
+  @Override
+  public void reset() throws IOException {
+    wrappedStream.reset();
+  }
+
+  @Override
+  public void mark(int readLimit) {
+    wrappedStream.mark(readLimit);
+  }
+
+  @Override
+  public void close() throws IOException {
```

Comment:
I don't see a test that verifies that the response gets closed if we close the stream.  Can you point me to it, or add one if it doesn't exist?

Reply:
I just added one test to verify that the response is closed when closing the content stream.

https://github.com/googleapis/google-http-java-client/blob/d8c7f67bd61131b41f6bdcd500818b4a5ce59f3d/google-http-client-apache-v5/src/test/java/com/google/api/client/http/apache/v5/Apache5HttpRequestTest.java#L126-L129